kernels/x86.py

```diff
@@ -45,17 +45,18 @@
 def is_imm(line): return '$' in line
 def is_memory(line): return '(' in line and 'lea' not in line and 'nop' not in line
 def is_mem_imm(line): return is_memory(line) and is_imm(line)
-def is_mem_load(line): return is_memory(line) and (is_type(LOAD_ANY, line) or C.any_in(('broadcast', 'insert', 'gather'), line))
-def is_mem_store(line): return is_memory(line) and is_type(STORE, line) and C.any_in(('mov', EXTRACT, 'scatter'), line)
+def is_mem_load(line): return is_memory(line) and (is_type(LOAD_ANY, line) or C.any_in(('broadcast', 'insert', 'gather'), line)) 
+def is_mem_store(line): return is_memory(line) and is_type(STORE, line) or C.any_in((EXTRACT, 'scatter'), line)
 def is_type(t, l): return re.match(r"\s+\S+\s+%s" % t, l) is not None
 def is_test_load(line): return is_type(CISC_CMP, line) or is_type(BIT_TEST, line)
 
 def get_mem_inst(line):
   assert '(' in line, line
+  store = is_mem_store(line)
   if 'lock' in line:        return 'lock'
   elif 'prefetch' in line:  return 'prefetch'
   elif is_test_load(line):  return 'load'
-  elif is_mem_store(line):  return 'store'
+  elif store and store == 'store':  return 'store'
   else: return 'load' if is_mem_load(line) else 'rmw'
 
```

Comment:
If you apply the change above then the new code here is not needed anymore, consider restoring the original code.

Reply:
I will revert to previous code